src/Components/Actions/ActionsTabItem.js

```diff
@@ -14,19 +17,43 @@ export default function ActionsTabItem({ actionItem, index }) {
     />
   );
 
-  const actionEndingTime = Math.floor(actionItem.endingTime / 100000);
-  //actionItem.endingTime is a unix timestamp regarding it's end,
-  //the division and floor are just for the large int to be small enough to fit in the screen
-  //Need to create dynamic counter for this and replace it.
+
+  // * When ending time is updated, set the format in seconds.
+  useEffect(() => {
+    if(typeof actionItem.endingTime === 'number' && typeof actionItem.startingTime === 'number')
+    {
+      let dur = moment.duration(moment(actionItem.endingTime).diff(moment(actionItem.startingTime)))
+      setActionDuration(dur.asSeconds().toFixed(0));
+    }
+  }, [actionItem])
+
+
+  // * set countdown for an action.
+  useEffect(() => {
+
+    // * if countdown get to 0, stop counting down.
+    // * In this part you can call the trigger of ending an action.
+    //TODO trigger updating server
+    if (actionDuration<=0) return;
+
+    // * set interval, every 1 second reduce the timer.
+    const intervalDuration = setInterval(() => {
+      setActionDuration(actionDuration - 1);
+    }, 1000);
+
+    // * clear the interval on cleanup.
+    return () => clearInterval(intervalDuration);
+
+  }, [actionDuration]);
 
   return (
-    <Draggable draggableId={actionItem.id} index={index}>
+    <Draggable id={actionItem.id} draggableId={actionItem.id} index={index}>
       {provided => (
         <div className="actionsTabItem" {...provided.draggableProps} ref={provided.innerRef}>
           <img src={hamburgerMenuIcon} alt="" className="smallIcons" {...provided.dragHandleProps} />
           <div>{actionItem.title}</div>
           {' - '}
-          <div>{actionEndingTime}</div>
+          <div>{actionDuration}</div>
```

Comment:
please display the duration in a "HH:MM:SS" format.
(preferably make things like 00:35:47 to automatically become 35:47)

Reply:
i decided to write custom hook for the timer. I'm working on that.